tfjs-layers/src/layers/nlp/modeling/transformer_decoder.ts

```diff
@@ -212,22 +250,84 @@ export class TransformerDecoder extends Layer {
    *  [decoderSequenceShape, encoderSequenceShape]
    */
   override build(inputShape: Shape|[Shape, Shape]): void {
-    throw new NotImplementedError(`Not implemented yet.`);
+    if (Array.isArray(inputShape[0])) {
+      // `inputShape` is of type [Shape, Shape].
+      [this.decoderSequenceShape, this.encoderSequenceShape] =
+        inputShape as [Shape, Shape];
+    } else {
+      this.decoderSequenceShape = inputShape as Shape;
+    }
+    // Infer the dimension of our hidden feature size from the build shape.
+    const hiddenDim =
+      this.decoderSequenceShape[this.decoderSequenceShape.length - 1];
+    // Attention head size is `hiddenDim` over the number of heads.
+    const headDim = Math.floor(hiddenDim / this.numHeads);
+
+    // Self attention layers.
+    this.selfAttentionLayer = new CachedMultiHeadAttention({
+      numHeads: this.numHeads,
+      keyDim: headDim,
+      dropout: this.dropout,
+      kernelInitializer: getInitializer(this.kernelInitializer.getClassName()),
+      biasInitializer: getInitializer(this.biasInitializer.getClassName()),
+    });
+
+    this.selfAttentionLayer.buildFromSignature(
+      this.decoderSequenceShape, this.decoderSequenceShape);
+
+    this.selfAttentionLayernorm =
+      new LayerNormalization({epsilon: this.layerNormEpsilon});
+
+    this.selfAttentionLayernorm.build(this.decoderSequenceShape);
+    this.selfAttentionDropout = new Dropout({rate: this.dropout});
+
+    // Cross attention layers are optional.
+    // TODO(pforderique): Add cross attention layers.
```

Comment:
As you mentioned here, we can skip cross attention since gpt2 doesn't require it.